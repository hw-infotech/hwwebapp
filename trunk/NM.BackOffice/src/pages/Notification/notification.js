import React from "react";
import ReCAPTCHA from "react-google-recaptcha";

const Notification = () => {
  function onChange(value) {
    console.log("Captcha value:", value);
  }
  function onSubmit(token) {
    document.getElementById("demo-form").submit();
  }
  // let a = [1, 2, 1, 5, 6, 4, 5, 5];
  // var filteredArray = a.filter(function (item, pos) {
  //   return a.indexOf(item) == pos;
  // });
  return (
    <div className="notification_content mt-2" id="demo-form">
      <h3>
        Notification <i class="fs-3 bi-bell"></i>
      </h3>

      <div className="notification_panel mt-4">
        <div className="notification_inner_part">
          <img
            src="./assets/images/NM-ICON.png"
            className="img-decorate"
            height="48px"
            width="48px"
          />
          <div>
            <p>John doe React your post</p>
            <p className="text-muted">
              Lorem ipsum dolor sit amet consectetur, adipisicing elit. Unde,
              dolorem.
            </p>
            <p className="text-muted">
              <small>10 minutes ago</small>
            </p>
          </div>
        </div>
        <div>
          <img
            src="./assets/images/ava3.png"
            className="img-decorate"
            width={"48px"}
            height={"48px"}
          />
        </div>
      </div>
      <div className="notification_panel mt-4">
        <div className="notification_inner_part">
          <img
            src="./assets/images/team_nestormind.png"
            className="img-decorate"
            height="48px"
            width="48px"
          />
          <div>
            <p>Richard Miles liked your post</p>
            <p className="text-muted">
              Lorem ipsum dolor sit amet consectetur, adipisicing elit. Unde,
              dolorem.
            </p>
            <p className="text-muted">
              <small>10 minutes ago</small>
            </p>
          </div>
        </div>
        <div>
          <img
            src="./assets/images/ava3.png"
            className="img-decorate"
            width={"48px"}
            height={"48px"}
          />
        </div>
      </div>
      <div className="notification_panel mt-4 mb-4">
        <div className="notification_inner_part">
          <img
            src="./assets/images/NM-ICON.png"
            className="img-decorate"
            height="48px"
            width="48px"
          />
          <div>
            <p>John doe React your post</p>
            <p className="text-muted">
              Lorem ipsum dolor sit amet consectetur, adipisicing elit. Unde,
              dolorem.
            </p>
            <p className="text-muted">
              <small>10 minutes ago</small>
            </p>
          </div>
        </div>
        <div>
          <img
            src="./assets/images/diwaliCel.jpg"
            className="img-decorate"
            width={"48px"}
            height={"48px"}
          />
        </div>
      </div>
      <ReCAPTCHA
        sitekey="6LckuYchAAAAAAwZOplPvCfxH2XXNRu9NsUXaaQ4"
        onChange={onChange}
      />
      <div
        class="g-recaptcha"
        data-sitekey="6LckuYchAAAAAAwZOplPvCfxH2XXNRu9NsUXaaQ4"
        data-callback="onSubmit"
        data-action="submit"
      >
        dsd
      </div>
    </div>
  );
};
export default Notification;
