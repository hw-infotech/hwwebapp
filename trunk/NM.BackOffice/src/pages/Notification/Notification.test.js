import {
  fireEvent,
  getAllByRole,
  getByTestId,
  getByText,
  render,
  screen,
  within,
} from "@testing-library/react";
import Notification from "./notification";
import renderer from "react-test-renderer";

describe("Testing the Notification", () => {
  test("Notification", () => {
    render(<Notification />);
    const items = screen.findAllByRole("button");
  });
  it("render all", () => {
    const tree = renderer.create(<Notification />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
