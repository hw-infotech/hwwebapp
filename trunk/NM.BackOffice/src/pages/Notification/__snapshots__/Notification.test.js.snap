// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the Notification render all 1`] = `
<div
  className="notification_content "
  id="demo-form"
>
  <div
    className="breadecumbs_box "
    onClick={[Function]}
  >
    <ul
      className="breadcumbs_style"
    >
      <li
        onClick={[Function]}
      >
        Dashboard
         
        &gt;
      </li>
      <li
        onClick={[Function]}
      >
        Notification
         
        &gt;
      </li>
      <li
        onClick={[Function]}
        style={
          Object {
            "color": "#737ed2",
          }
        }
      >
        View all
      </li>
    </ul>
  </div>
  <div
    className="panle_body"
  >
    <div
      className="panle_header"
    >
      <div
        className="left-panle-title"
      >
        <h4>
          Notification
        </h4>
      </div>
      <div
        className="right_panle_container"
      >
        <button
          className="btn-sm remove_button_padding btn"
          data-testid="notifi"
          disabled={false}
          onClick={[Function]}
          title="Filter"
          type="button"
        >
          <svg
            color="#ff6b01"
            fill="currentColor"
            height={24}
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": "#ff6b01",
              }
            }
            viewBox="0 0 16 16"
            width={24}
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M6 10.5a.5.5 0 0 1 .5-.5h3a.5.5 0 0 1 0 1h-3a.5.5 0 0 1-.5-.5zm-2-3a.5.5 0 0 1 .5-.5h7a.5.5 0 0 1 0 1h-7a.5.5 0 0 1-.5-.5zm-2-3a.5.5 0 0 1 .5-.5h11a.5.5 0 0 1 0 1h-11a.5.5 0 0 1-.5-.5z"
            />
          </svg>
        </button>
      </div>
    </div>
    <div
      className="gapbetween pt-1"
    >
      <div
        aria-expanded={null}
        className="custom-date-box mb-3 collapse"
      >
        <input
          className="form-control custom-date-filter"
          name="date"
          type="date"
        />
      </div>
    </div>
  </div>
  <div
    className="notification_panel "
  >
    <div
      className="notification_inner_part"
    >
      <img
        className="img-decorate"
        height="30px"
        src="./assets/images/NM-ICON.png"
        width="30px"
      />
      <div>
        <p>
          John doe React your post
        </p>
        <p
          className="text-muted"
        >
          Lorem ipsum dolor sit amet consectetur, adipisicing elit. Unde, dolorem.
        </p>
        <p
          className="text-muted"
        >
          <small>
            10 minutes ago
          </small>
        </p>
      </div>
    </div>
    <div>
      <img
        className="img-decorate"
        height="30px"
        src="./assets/images/ava3.png"
        width="30px"
      />
    </div>
  </div>
  <div
    className="notification_panel mt-4"
  >
    <div
      className="notification_inner_part"
    >
      <img
        className="img-decorate"
        height="30px"
        src="./assets/images/team_nestormind.png"
        width="30px"
      />
      <div>
        <p
          style={
            Object {
              "color": "#004a87",
            }
          }
        >
          Richard Miles liked your post
        </p>
        <p
          className="text-muted"
        >
          Lorem ipsum dolor sit amet consectetur, adipisicing elit. Unde, dolorem.
        </p>
        <p
          className="text-muted"
        >
          <small>
            10 minutes ago
          </small>
        </p>
      </div>
    </div>
    <div>
      <img
        className="img-decorate"
        height="30px"
        src="./assets/images/ava3.png"
        width="30px"
      />
    </div>
  </div>
  <div
    className="notification_panel mt-4 mb-4"
  >
    <div
      className="notification_inner_part"
    >
      <img
        className="img-decorate"
        height="30px"
        src="./assets/images/NM-ICON.png"
        width="30px"
      />
      <div>
        <p>
          John doe React your post
        </p>
        <p
          className="text-muted"
        >
          Lorem ipsum dolor sit amet consectetur, adipisicing elit. Unde, dolorem.
        </p>
        <p
          className="text-muted"
        >
          <small>
            10 minutes ago
          </small>
        </p>
      </div>
    </div>
    <div>
      <img
        className="img-decorate"
        height="30px"
        src="./assets/images/diwaliCel.jpg"
        width="30px"
      />
    </div>
  </div>
</div>
`;
